habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ValidatePythonCodeMojo.java

```diff
@@ -0,0 +1,135 @@
+package org.technologybrewery.habushu;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.technologybrewery.habushu.exec.PoetryCommandHelper;
+
+/**
+ * Leverages the lint package to validate both source and test Python
+ * directories using Poetry's run command.
+ */
+@Mojo(name = "validate-python-code", defaultPhase = LifecyclePhase.PROCESS_CLASSES, requiresDependencyResolution = ResolutionScope.COMPILE)
+public class ValidatePythonCodeMojo extends AbstractHabushuMojo {
+
+    protected static final String LINT_PACKAGE = "pylint";
+
+    /**
+     * Specifies disabled checkers for lint on source module.
+     */
+    @Parameter(property = "habushu.sourceLintDisabledChecker", required = false, defaultValue = "C,R,W")
+    private String sourceLintDisabledChecker;
+
+    /**
+     * Specifies disabled checkers for lint on test module.
+     */
+    @Parameter(property = "habushu.testLintDisabledChecker", required = false, defaultValue = "C,R,W")
+    private String testLintDisabledChecker;
+
+    /**
+     * Specifies enabled checkers for lint on source module.
+     */
+    @Parameter(property = "habushu.sourceLintEnabledChecker", required = false)
+    private String sourceLintEnabledChecker;
+
+    /**
+     * Specifies enabled checkers for lint on test module.
+     */
+    @Parameter(property = "habushu.testLintEnabledChecker", required = false)
+    private String testLintEnabledChecker;
+
+    /**
+     * By default, build will stop if lint errors are found in source module. Can be configured to true so that
+     * build will continue. 
+     */
+    @Parameter(property = "habushu.sourceFailOnLintErrors", required = false, defaultValue = "false")
+    private boolean sourceFailOnLintErrors;
+
+    /**
+     * By default, build will stop if lint errors are found in test module. Can be configured to true so that
+     * build will continue. 
+     */
+    @Parameter(property = "habushu.testFailOnLintErrors", required = false, defaultValue = "false")
+    private boolean testFailOnLintErrors;
+
+
+
+    @Override
+    public void doExecute() throws MojoExecutionException {
+        PoetryCommandHelper poetryHelper = createPoetryCommandHelper();
+
+        List<String> sourceDirectoriesToValidate = new ArrayList<>();
+        if (this.sourceDirectory.exists()) {
+            sourceDirectoriesToValidate.add(getCanonicalPathForFile(sourceDirectory));
+        }
+
+        if (sourceDirectoriesToValidate.isEmpty()) {
+            getLog().warn(String.format("Configured source (%s) directories does not exist - skipping...",
```

Comment:
A:
```suggestion
            getLog().warn(String.format("Configured source (%s) directories do not exist - skipping...",
```

Reply:
This pattern was recognized from the FormatPythonMojo.java. I've made the changes to clean up the code, but we should flow another PR to address FormatPythonMojo as well.